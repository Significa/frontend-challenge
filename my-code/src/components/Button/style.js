import sc, { css } from 'styled-components'
import { theme } from 'styled-tools'

import { text4 } from '../../styles/text'

export const Wrapper = sc.button`
  display: flex;
  align-items: center;
  /* Account for border */
  padding: 0 15px 0 11px;
  height: 40px;
  outline: none;
  background: none;
  color: ${theme('greyLight')};
  border: 1px solid ${theme('greyMid')};
  border-radius: 4px;
  cursor: pointer;
  transition-timing-function: ${theme('transitionTiming')};
  transition-duration: ${theme('transitionDuration')};
  transition-property: color, border-color, background-color;
  font-family: inherit;
  ${text4}
  font-weight: 500;

  &:focus,
  &:hover {
    border-color: ${theme('mainRed')};
    color: ${theme('greyWhite')};
  }

  ${({ active }) =>
    active &&
    css`
      border-color: ${theme('mainRed')};
      background-color: ${theme('mainRed')};
      color: ${theme('greyWhite')};
    `}
`

export const IconWrapper = sc.div`
  width: 16px;
  height: 16px;
  margin-right: 12px;
  display: flex;
  align-items: center;
`
