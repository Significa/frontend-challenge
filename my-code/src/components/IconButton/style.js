import sc from 'styled-components'
import { theme } from 'styled-tools'

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = sc.button`
  height: 40px;
  outline: none;
  border: none;
  padding: 0;
  background: none;
  color: ${theme('greyLight')};
  cursor: pointer;
  transition:
    color
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;
  display: flex;
  align-items: center;
  justify-content: center;

  &:focus,
  &:hover {
    color: ${theme('greyWhite')};
  }
`
