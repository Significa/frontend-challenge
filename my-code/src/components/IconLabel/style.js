import sc from 'styled-components'
import { theme, prop } from 'styled-tools'

import { text4 } from '../../styles/text'

export const Wrapper = sc.div`
  display: flex;
  align-items: stretch;
  height: 40px;
  margin-right: 16px;
  margin-bottom: 8px;
`

export const IconWrapper = sc.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${prop('color')};
  border-radius: 4px 0 0 4px;
  padding: 8px;
  min-width: 40px;
  box-sizing: border-box;
`

export const TextWrapper = sc.div`
  border: 1px solid ${theme('greyMid')};
  border-left: none;
  /* Account for border */
  padding: 8px 7px 8px 8px;
  border-radius: 0 4px 4px 0;
  ${text4}
  font-weight: 500;
`
