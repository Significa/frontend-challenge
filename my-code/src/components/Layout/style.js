import sc from 'styled-components'

// eslint-disable-next-line import/prefer-default-export
export const Outer = sc.div`
  display: flex;
  flex-direction: column;
  max-width: 1228px;
  margin: 0 auto;
  padding: 24px;
  min-height: 100vh;
  box-sizing: border-box;
`
