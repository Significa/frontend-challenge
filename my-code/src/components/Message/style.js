import sc from 'styled-components'
import { theme } from 'styled-tools'

import { text2, text4 } from '../../styles/text'

import illustration from '../../assets/illustration-empty-state@2x.png'

export const Center = sc.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
`

export const Illustration = sc.img.attrs({
  src: illustration,
  alt: ''
})`
  object-fit: contain;
  width: 400px;
  max-width: 100%;
  height: 200px;
`

export const Title = sc.span`
  ${text2}
  margin-top: 16px;
`

export const Subtitle = sc.span`
  ${text4}
  color: ${theme('greyLight')};
  margin-top: 8px;
`
