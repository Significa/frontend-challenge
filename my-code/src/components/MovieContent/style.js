import sc from 'styled-components'
import { theme } from 'styled-tools'

import { text1, text3, text4 } from '../../styles/text'

export const Wrapper = sc.div`
  margin-top: 24px;
  display: grid;
  grid-gap: 24px 96px;
  grid-template-columns: 0.6fr 0.4fr;

  @media (max-width: 1024px) {
    grid-column-gap: 24px;
    grid-template-columns: auto minmax(30vw, 0.4fr);
  }

  @media (max-width: 700px) {
    grid-template-columns: 1fr;
  }
`

export const Row = sc.div`
  grid-column: 1;
  display: flex;
  flex-wrap: wrap;
  flex-direction: ${({ direction }) => direction || 'row'};
`

export const MetaText = sc.span`
  ${text3}
  font-weight: 400;
  color: ${theme('greyLight')};
`

export const Title = sc.span`
  ${text1}
  display: block;
  max-width: 20ch;
  margin: 0 0 8px;

  @media (max-width: 1024px) {
    font-size: 56px;
  }

  @media (max-width: 800px) {
    font-size: 48px;
  }
`

export const Text = sc.span`
  ${text4}
`

export const Heading = sc.span`
  display: block;
  margin-bottom: 8px;
  ${text4}
  font-weight: 500;
  color: ${theme('greyLight')};

  margin-top: 16px;
`

export const PosterWrapper = sc.div`
  grid-column: 2;
  grid-row: 2 / 7;

  position: relative;

  @media (max-width: 1000px) {
    grid-row: 2 / 4;
  }

  @media (max-width: 700px) {
    display: none;
  }
`

export const Poster = sc.img`
  border-radius: 8px;

  position: absolute;
  top: 0;
  right: 0;
  max-width: 100%;
`

export const Paragraph = sc.p`
  max-width: 54ch;
  margin: 0;
`

export const Subgrid = sc.div`
  grid-column: 1;
  display: grid;
  grid-gap: 56px;
  grid-template-columns: repeat(3, minmax(0, max-content));

  @media (max-width: 480px) {
    grid-template-columns: none;
    grid-gap: 24px;
  }
`
