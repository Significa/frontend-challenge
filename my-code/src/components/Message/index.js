import React from 'react'
import PropTypes from 'prop-types'

import { Center, Illustration, Title, Subtitle } from './style'

export default function Message({ title, subtitle }) {
  return (
    <Center>
      <Illustration />
      {title && <Title>{title}</Title>}
      {subtitle && <Subtitle>{subtitle}</Subtitle>}
    </Center>
  )
}

Message.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string
}

Message.defaultProps = {
  title: undefined,
  subtitle: undefined
}
