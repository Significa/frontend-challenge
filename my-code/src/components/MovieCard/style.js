import sc from 'styled-components'
import { Link } from 'react-router-dom'
import { theme } from 'styled-tools'

import { text3, text4 } from '../../styles/text'
import IconButton from '../IconButton'

export const Wrapper = sc.div`
  width: 100%;
  text-decoration: none;
  position: relative;
  border-radius: 3px;
  overflow: hidden;
  color: ${theme('greyWhite')};

  &:focus {
    outline: none;
  }

  /* Aspect ratio */
  &:before {
    content: '';
    display: block;
    padding-top: 133.3333333%;
  }
`

export const CoverLink = sc(Link)`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`

export const Poster = sc.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
  background-color: ${theme('greyMain')};
  background-image: ${({ poster }) => `url(${poster})`};
`

export const FavoriteButtonWrapper = sc(IconButton)`
  height: auto;
  position: absolute;
  z-index: 1;
  top: 12px;
  right: 12px;
  color: ${theme('greyWhite')};
  will-change: opacity;
  transition:
    opacity
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;
  filter: drop-shadow(0 0 2px rgba(0, 0, 0, .2));
  opacity: ${({ visible = true }) => (visible ? 1 : 0)};

  ${Poster} ~ ${CoverLink}:hover ~ &,
  ${Poster} ~ ${CoverLink}:focus ~ &,
  ${Poster} ~ &:hover,
  ${Poster} ~ &:focus {
    opacity: 1;
  }
`

export const Title = sc.span`
  ${text3}
`

export const Subtitle = sc.span`
  ${text4}
`

export const Content = sc.div`
  pointer-events: none;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  padding: 12px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background: ${theme('scrimColor')};
  will-change: opacity;
  transition:
    opacity
    ${theme('transitionTiming')}
    ${theme('transitionDuration')}
  ;

  ${Poster} ~ & {
    opacity: 0;
  }

  ${Poster} ~ ${CoverLink}:hover ~ &,
  ${Poster} ~ ${CoverLink}:focus ~ &,
  ${Poster} ~ ${FavoriteButtonWrapper}:hover ~ &,
  ${Poster} ~ ${FavoriteButtonWrapper}:focus ~ & {
    opacity: 1;
  }
`
