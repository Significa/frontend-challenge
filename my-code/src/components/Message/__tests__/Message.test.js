import React from 'react'
import { shallow } from 'enzyme'

import Message from '../'
import { Center, Illustration, Title, Subtitle } from '../style'

describe('Message', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <Message title="message title" subtitle="message subtitle" />
    )
    expect(tree).toMatchSnapshot()
  })

  describe('Center', () => {
    it('renders correctly', () => {
      const tree = shallow(<Center />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Illustration', () => {
    it('renders correctly', () => {
      const tree = shallow(<Illustration />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Title', () => {
    it('renders correctly', () => {
      const tree = shallow(<Title>title text</Title>)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Subtitle', () => {
    it('renders correctly', () => {
      const tree = shallow(<Subtitle>subtitle text</Subtitle>)
      expect(tree).toMatchSnapshot()
    })
  })
})
