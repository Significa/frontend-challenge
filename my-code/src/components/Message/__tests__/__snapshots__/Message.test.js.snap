// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Center renders correctly 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

<div
  className="c0"
/>
`;

exports[`Message Illustration renders correctly 1`] = `
.c0 {
  object-fit: contain;
  width: 400px;
  max-width: 100%;
  height: 200px;
}

<img
  alt=""
  className="c0"
  src="image-mock-url"
/>
`;

exports[`Message Subtitle renders correctly 1`] = `
.c0 {
  font-weight: 400;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  color: theme.greyLight;
  margin-top: 8px;
}

<span
  className="c0"
>
  subtitle text
</span>
`;

exports[`Message Title renders correctly 1`] = `
.c0 {
  font-weight: 500;
  font-size: 24px;
  line-height: 1.25em;
  -webkit-letter-spacing: 0.008333333333333333em;
  -moz-letter-spacing: 0.008333333333333333em;
  -ms-letter-spacing: 0.008333333333333333em;
  letter-spacing: 0.008333333333333333em;
  margin-top: 16px;
}

<span
  className="c0"
>
  title text
</span>
`;

exports[`Message renders correctly 1`] = `
<Center>
  <Illustration />
  <Title>
    message title
  </Title>
  <Subtitle>
    message subtitle
  </Subtitle>
</Center>
`;
